### PR TITLE
KYLIN-4015 "Build UHC Dictionary" step filter ".dci" files 

### PR DESCRIPTION
**Pull request by zhaojintaozhao**
KYLIN-4015 "Build UHC Dictionary" step filter ".dci" files to solve the problem that MR engine run failed because the ".dci" file is  not a  sequence file.

### CODE REVIEW

**Comment by asfgit** [comment]
Can one of the admins verify this patch?

File: engine-mr/src/main/java/org/apache/kylin/engine/mr/steps/filter/UHCDictPathFilter.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+    @Override
+    public boolean accept(Path path) {
+
+        Pattern pattern = Pattern.compile(DCIFILE_POSTFIX);
+        Matcher matcher = pattern.matcher(path.getName().toLowerCase(Locale.ROOT));
+
+        if (matcher.find()) {
+            logger.info("filter file: " + path.getName());
+            return false;
```
[the remark this answers is not in the document]
Comment (zhaojintaozhao) [reply engine-mr/src/main/java/org/apache/kylin/engine/mr/steps/filter/UHCDictPathFilter.java]: Hi Shaofeng,  That's a good idea, I will submit new  code and filter in a  whitelist way.

File: engine-mr/src/main/java/org/apache/kylin/engine/mr/steps/filter/UHCDictPathFilter.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+import org.slf4j.LoggerFactory;
+
+import java.util.Locale;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class UHCDictPathFilter implements PathFilter {
+    private static final Logger logger = LoggerFactory.getLogger(UHCDictPathFilter.class);
+
+    private static final String DCIFILE_POSTFIX = ".dci";
```
[the remark this answers is not in the document]
Comment (zhaojintaozhao) [reply engine-mr/src/main/java/org/apache/kylin/engine/mr/steps/filter/UHCDictPathFilter.java]: Hi Shaofeng, I will submit new code and use a common place. Thank's for your review.

File: engine-mr/src/main/java/org/apache/kylin/engine/mr/steps/filter/UHCDictPathFilter.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (zhaojintaozhao) [reply engine-mr/src/main/java/org/apache/kylin/engine/mr/steps/filter/UHCDictPathFilter.java]: Hi Shaofeng
How can we implement it with a whitelist way? After "Extract Fact Table Distinct Columns" step, Kylin generate files  like "colName.dci-r-00001", "colName.rldict-r-00002" or "-r-00002","-r-00003".  After filtering path,   only file name like ".rldict-r-00001" or "-r-00002" are retained?
I want to check whether the type of hdfs file is "SequenceFile", but I don't find any hadoop API can provide this ability. Hadoop does not perceive the specific type of file.

File: engine-mr/src/main/java/org/apache/kylin/engine/mr/steps/filter/UHCDictPathFilter.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (zhaojintaozhao) [reply engine-mr/src/main/java/org/apache/kylin/engine/mr/steps/filter/UHCDictPathFilter.java]: Hi Shaofeng:
I have replaced ".dci" to FactDistinctColumnsReducer.DIMENSION_COL_INFO_FILE_POSTFIX in my latest PR.
Thank you.

**Pull request merged**